### ISSUE TITLE
Missing "./webgpu" specifier in "three" package

### ISSUE THREAD

**Issue by FlipFloop** [mrdoob/three.js#29990]
### Description

Upon upgrading, vercel builds won't work anymore.

I am using astrojs, with a react island.

Here is the error: ` x Build failed in 12.74s
[commonjs--resolver] Missing "./webgpu" specifier in "three" package`

### Reproduction steps

1. Upgrade pnpm/npm
2. deploy to vercel

### Code

```json
"dependencies": {
    "@astrojs/check": "^0.4.1",
    "@astrojs/mdx": "^2.3.1",
    "@astrojs/react": "^3.6.3",
    "@astrojs/sitemap": "^3.2.1",
    "@astrojs/vercel": "^6.1.4",
    "@iconify-json/fa-solid": "^1.2.0",
    "@mojs/core": "^1.7.1",
    "@mojs/player": "^1.3.0",
    "@playform/compress": "^0.0.3",
    "@react-three/drei": "^9.117.3",
    "@react-three/fiber": "^8.17.10",
    "@react-three/offscreen": "^0.0.8",
    "@react-three/postprocessing": "^2.16.3",
    "@tsparticles/engine": "^3.7.1",
    "@types/react": "^18.3.12",
    "@types/react-dom": "^18.3.1",
    "@types/three": "^0.160.0",
    "astro": "^4.16.16",
    "astro-font": "^0.0.71",
    "astro-icon": "^1.1.4",
    "astro-particles": "^2.10.0",
    "astro-robots": "^2.2.0",
    "csv-parse": "^5.6.0",
    "d3": "^7.9.0",
    "highcharts": "^11.4.8",
    "lamina": "^1.1.23",
    "leaflet": "^1.9.4",
    "react": "^18.3.1",
    "react-dom": "^18.3.1",
    "react-globe.gl": "^2.27.6",
    "rough-notation": "^0.5.1",
    "simsimd": "^3.9.0",
    "three": "^0.160.1",
    "tsparticles": "^3.7.1"
  },
  "devDependencies": {
    "@biomejs/biome": "^1.9.4",
    "@types/d3": "^7.4.3",
    "astro-robots-txt": "^1.0.0",
    "astro-seo": "^0.8.4",
    "autoprefixer": "^10.4.20",
    "cssnano": "^6.1.2",
    "prettier": "^3.4.1",
    "prettier-plugin-astro": "^0.12.3",
    "sass": "^1.81.0",
    "typescript": "^5.7.2"
  },
  "packageManager": "pnpm@8.14.1+sha256.2df78e65d433d7693b9d3fbdaf431b2d96bb4f96a2ffecd51a50efe16e50a6a8"

```

```jsx
import { Canvas } from "@react-three/offscreen";
import React, { lazy, Suspense } from "react";

// This is the fallback component that will be rendered on the main thread
// This will happen on systems where OffscreenCanvas is not supported
const Scene = lazy(() => import("./Scene"));

// This is the worker thread that will render the scene
const worker = new Worker(new URL("./worker.jsx", import.meta.url), {
  type: "module",
});

export const Main = () => {
  return (
    <Suspense fallback={<h2 style={"color: white"}>Loading...</h2>}>
      <Canvas
        camera={{ position: [0, 0, 10], fov: 25 }}
        worker={worker}
        fallback={<Scene />}
      />
    </Suspense>
  );
};

```

```jsx
import {
  CameraControls,
  Center,
  ContactShadows,
  Environment,
  useGLTF,
} from "@react-three/drei";
import { useFrame } from "@react-three/fiber";
import React, { useState, useRef } from "react";

import { MathUtils } from "three";

import { Blob } from "./Blob";

const Model = () => {
  const mesh = useRef();
  const { nodes, materials } = useGLTF("/assets/pmndrs.glb");
  const [hovered, setHover] = useState(false);
  const [active, setActive] = useState(false);
  const color = hovered ? "hotpink" : "orange";
  useFrame((state, delta) => {
    mesh.current.rotation.x += delta / 2;
    mesh.current.rotation.y += delta / 2;
  });
  return (
    <>
      <Center ref={mesh}>
        <mesh
          geometry={nodes.cube.geometry}
          material={materials.base}
          material-color={color}
          scale={active ? 0.3 : 0.25}
          onClick={(e) => (e.stopPropagation(), setActive(!active))}
          onPointerOver={(e) => (e.stopPropagation(), setHover(true))}
          onPointerOut={(e) => setHover(false)}
        />
      </Center>
      <ContactShadows
        color={color}
        position={[0, -1.5, 0]}
        blur={3}
        opacity={0.75}
      />
    </>
  );
};

const Scene = () => {
  const getRandCoords = () => {
    const xPos = MathUtils.randFloat(-2, 4);
    const yPos = MathUtils.randFloat(-0.5, 2.5);
    const zPos = MathUtils.randFloat(-2, 4);
    return [xPos, yPos, zPos];
  };

  return (
    <>
      <ambientLight />
      <pointLight position={[10, 10, 5]} />
      <Center>
        <Model />
        <Blob position={getRandCoords()} />
        <Blob position={getRandCoords()} />
        <Blob position={getRandCoords()} />
        <Blob position={getRandCoords()} />
        <Blob position={getRandCoords()} />
        <Blob position={getRandCoords()} />
        <Blob position={getRandCoords()} />
      </Center>

      {/* <Blob position={[-2, 1.5, 0]} />
      <Blob position={[-2, 0.5, 0]} />
      <Blob position={[-1, 0.7, 0]} />
      <Blob position={[-2, 0.6, 0]} />
      <Blob position={[2, 0.8, 1]} />
      <Blob position={[-2, -0.2, 4]} />
      <Blob position={[2, 1, -2]} /> */}
      <Environment preset="city" />
      <CameraControls
        minPolarAngle={Math.PI / 2}
        maxPolarAngle={Math.PI / 2}
        maxDistance={10}
        minDistance={10}
      />
      {/* <EffectComposer multisampling={4}>
        <DepthOfField target={[0, 0, 60]} focalLength={0.4} bokehScale={14} height={700} />
      </EffectComposer> */}
    </>
  );
};

export default Scene;

```

### Live example

* [jsfiddle-latest-release WebGLRenderer](https://jsfiddle.net/3mrkqyea/)
* [jsfiddle-dev WebGLRenderer](https://jsfiddle.net/gcqx26jv/)
* [jsfiddle-latest-release WebGPURenderer](https://jsfiddle.net/8L2jkmx7/)
* [jsfiddle-dev WebGPURenderer](https://jsfiddle.net/L3n1w4yh/)

### Screenshots

<img width="1165" alt="Screenshot 2024-11-28 at 16 37 38" src="https://github.com/user-attachments/assets/7c3bbd7a-f74c-436e-a96e-3e9ad4671ac0">

### Version

0.160.0

### Device

Desktop

**Comment by Mugen87** [comment]
> "three": "^0.160.1",

Do you mind upgrading to the latest version `r170` and see if the issue still persists?

**Comment by FlipFloop** [comment]
Gone.
I guess React-globe should update their deps.
Easy. Can't believe I didn't do this earlier. Happy thanksgiving!
Thank you!

**Comment by robsongajunior** [comment]
I have the same error with vue

**Comment by vladinator1000** [comment]
Getting this error on three `v0.174.0`. I use Three as a transitive dependency:

```
 node_modules/3d-force-graph/node_modules/three-render-objects/dist/three-render-objects.mjs:2:31:
      2 │ import { WebGPURenderer } from 'three/webgpu';
```

**Comment by Bug-Reaper** [comment]
Experienced error with `Astro/Svelte` and fixed via upgrade to `v0.175.0` 🙌
